src/stp/new_plays/Defend.cpp

```diff
@@ -2,74 +2,138 @@
+        std::make_unique<role::Formation>(role::Formation("offender_1")),
+        std::make_unique<role::Formation>(role::Formation("offender_2"))};
 }
 
-uint8_t Defend::score(world_new::World* world) noexcept { return 12; }
+uint8_t Defend::score(world_new::World* world) noexcept { return 50; }
 
 Dealer::FlagMap Defend::decideRoleFlags() const noexcept {
     Dealer::FlagMap flagMap;
-    Dealer::DealerFlag closeToBallFlag(DealerFlagTitle::CLOSE_TO_BALL, DealerFlagPriority::HIGH_PRIORITY);
-    Dealer::DealerFlag closeToTheirGoalFlag(DealerFlagTitle::CLOSE_TO_THEIR_GOAL, DealerFlagPriority::MEDIUM_PRIORITY);
-    Dealer::DealerFlag notImportant(DealerFlagTitle::CLOSE_TO_OUR_GOAL, DealerFlagPriority::LOW_PRIORITY);
-
-    flagMap.insert({"defender_1", {closeToBallFlag}});
-    flagMap.insert({"defender_2", {closeToTheirGoalFlag}});
-    flagMap.insert({"test_role_2", {notImportant}});
-    flagMap.insert({"test_role_3", {closeToTheirGoalFlag}});
-    flagMap.insert({"test_role_4", {closeToBallFlag}});
-    flagMap.insert({"test_role_5", {closeToTheirGoalFlag, closeToBallFlag}});
-    flagMap.insert({"test_role_6", {closeToBallFlag}});
-    flagMap.insert({"test_role_7", {closeToTheirGoalFlag}});
-    flagMap.insert({"test_role_8", {closeToTheirGoalFlag, closeToBallFlag}});
-    flagMap.insert({"test_role_9", {closeToBallFlag}});
-    flagMap.insert({"test_role_10", {closeToTheirGoalFlag}});
+    Dealer::DealerFlag keeperFlag(DealerFlagTitle::KEEPER, DealerFlagPriority::REQUIRED);
+    Dealer::DealerFlag closeToOurGoalFlag(DealerFlagTitle::CLOSE_TO_OUR_GOAL, DealerFlagPriority::HIGH_PRIORITY);
+    Dealer::DealerFlag closeToTheirGoalFlag(DealerFlagTitle::CLOSE_TO_THEIR_GOAL, DealerFlagPriority::LOW_PRIORITY);
+
+    flagMap.insert({"keeper", {keeperFlag}});
+    flagMap.insert({"defender_1", {closeToOurGoalFlag}});
+    flagMap.insert({"defender_2", {closeToOurGoalFlag}});
+    flagMap.insert({"defender_3", {closeToOurGoalFlag}});
+    flagMap.insert({"defender_4", {closeToOurGoalFlag}});
+    flagMap.insert({"midfielder_1", {}});
+    flagMap.insert({"midfielder_2", {}});
+    flagMap.insert({"midfielder_3", {}});
+    flagMap.insert({"midfielder_4", {}});
+    flagMap.insert({"offender_1", {closeToTheirGoalFlag}});
+    flagMap.insert({"offender_2", {closeToTheirGoalFlag}});
 
     return flagMap;
 }
 
 void Defend::calculateInfoForRoles() noexcept {
+    calculateInfoForDefenders();
+    calculateInfoForKeeper();
+    calculateInfoForMidfielders();
+    calculateInfoForOffenders();
+}
+
+void Defend::calculateInfoForDefenders() noexcept {
+    auto enemyRobots = world->getWorld()->getThem();
+    auto enemyAttacker = world->getWorld()->getRobotClosestToBall(world_new::them);
+
+    for (auto enemyRobot = enemyRobots.begin(); enemyRobot < enemyRobots.end(); enemyRobot++) {
+        if (enemyRobot->get()->getId() == enemyAttacker->getId()) {
+            enemyRobots.erase(enemyRobot);
+        }
+    }
```

Comment:
I would use std::erase_if

Reply:
Oops, std::erase_if is C++20, you should use [this](https://stackoverflow.com/questions/39019806/using-erase-remove-if-idiom) instead. std::erase(std::remove_if())